src/js/view/view.js

```diff
@@ -868,14 +868,21 @@ function View(_api, _model) {
             _this.trigger(FLOAT, { floating: true });
             _model.set('floating', true);
 
-            _resizeOnFloat = true;
+            if (_isMobile) {
+                addClass(_wrapperElement, 'jw-float-top');
+            } else {
+                addClass(_wrapperElement, 'jw-float-bottom');
+                _resizeOnFloat = true;
 
-            // Resize within MAX_FLOATING_WIDTH×MAX_FLOATING_HEIGHT bounds, never enlarge.
-            const { width, height } = _this.getSafeRegion(false);
-            const ratio = Math.min(1, MAX_FLOATING_WIDTH / width, MAX_FLOATING_HEIGHT / height);
-            _this.resize(width * ratio, height * ratio, true);
+                // Resize within MAX_FLOATING_WIDTH×MAX_FLOATING_HEIGHT bounds, never enlarge.
+                const { width, height } = _this.getSafeRegion(false);
```

Comment:
I think we need unit tests here to clarify the logic. I'm a bit confused by the math - I'm going to work it out on paper but I think tests would be helpful.

Reply:
Test cases should be when both are under the max, both over, one over/one under, and reciprocal

Reply:
There is no reason to call `getSafeRegion` inside view when you have `_lastWidth` and `_lastHeight` with the values you are looking for.

Reply:
Thanks @vseventer